javaslang/src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -51,6 +51,88 @@
         PUT, REMOVE
     }
 
+    class LeafNodeIterator<K, V> extends AbstractIterator<LeafNode<K, V>> {
+
+        // buckets levels + leaf level = (Integer.SIZE / AbstractNode.SIZE + 1) + 1
+        private final static int MAX_LEVELS = Integer.SIZE / AbstractNode.SIZE + 2;
```

Comment:
I don't understand the comment.

`(Integer.SIZE / AbstractNode.SIZE + 1) + 1` not equals `Integer.SIZE / AbstractNode.SIZE + 2`